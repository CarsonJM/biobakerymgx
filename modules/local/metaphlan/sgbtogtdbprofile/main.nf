process METAPHLAN_SGBTOGTDBPROFILE {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/metaphlan:4.1.0--pyhca03a8a_0' :
        'biocontainers/metaphlan:4.1.0--pyhca03a8a_0' }"

    input:
    tuple val(meta), path(metaphlan_profile)
    path (sgb2gtdb)

    output:
    tuple val(meta), path("${prefix}_gtdb_profile.txt") , emit: gtdb_profile
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    metaphlan_sgb_to_gtdb_profile.py \\
        -i ${metaphlan_profile} \\
        -d ./${sgb2gtdb} \\
        -o ${prefix}_gtdb_profile.txt \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metaphlan: \$(metaphlan --version 2>&1 | awk '{print \$3}')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_gtdb_profile.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metaphlan: \$(metaphlan --version 2>&1 | awk '{print \$3}')
    END_VERSIONS
    """
}
