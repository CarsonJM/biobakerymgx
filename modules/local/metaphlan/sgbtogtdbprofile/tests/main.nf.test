nextflow_process {

    name "Test Process: METAPHLAN_SGBTOGTDBPROFILE"
    script "../main.nf"
    process "METAPHLAN_SGBTOGTDBPROFILE"

    tag "modules"
    tag "modules_local"
    tag "metaphlan"
    tag "metaphlan/sgbtogtdbprofile"

    test(" metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] ") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile1_txt'], checkifExists: true )
                    ]
                )
                input[1] = file( params.test_data['assets']['metaphlan']['metaphlan_sgb2gtdb_tsv'], checkifExists: true )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test(" metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] -stub ") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file( params.test_data['assets']['metaphlan']['metaphlan_sgb_profile1_txt'], checkifExists: true )
                    ]
                )
                input[1] = file( params.test_data['assets']['metaphlan']['metaphlan_sgb2gtdb_tsv'], checkifExists: true )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
