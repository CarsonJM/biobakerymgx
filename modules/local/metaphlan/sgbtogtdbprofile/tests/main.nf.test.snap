{
    " metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['local_modules']['metaphlan_sgb_profile_txt'] ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,5a3814d681d53102d64c5106285a11d1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ],
                "gtdb_profile": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,5a3814d681d53102d64c5106285a11d1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ]
            }
        ],
        "timestamp": "2024-01-24T08:18:10.244049698"
    },
    " metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['local_modules']['metaphlan_sgb_profile_txt'] -stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ],
                "gtdb_profile": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ]
            }
        ],
        "timestamp": "2024-01-24T08:18:23.490857699"
    },
    " metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] -stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ],
                "gtdb_profile": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ]
            }
        ],
        "timestamp": "2024-01-24T14:34:59.136322339"
    },
    " metaphlan_db_version = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103' & ['assets']['metaphlan']['metaphlan_sgb_profile1_txt'] ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,f0c993c440f0049ecd29d4925e730c39"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ],
                "gtdb_profile": [
                    [
                        {
                            "id": "test"
                        },
                        "test_gtdb_profile.txt:md5,f0c993c440f0049ecd29d4925e730c39"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e838b899e50a9a2e3b8d83288a8acb9d"
                ]
            }
        ],
        "timestamp": "2024-01-24T14:34:46.011418803"
    }
}